arc/settings.py

```diff
@@ -13,5 +13,10 @@ class Settings(pydantic.BaseSettings):
     board_gap: int = 5
     pair_gap: int = 1
 
-
 settings = Settings()
+
+CELL_PADDING_STR = " " * settings.cell_padding
+BOARD_GAP_STR = " " * settings.board_gap
+PAIR_GAP_STR = "\n" + " " * settings.pair_gap + "\n"
+DATASET_DIR = settings.dataset_dir
+CACHE_PATH = settings.cache_path
```

Comment:
is there a reason duplicating the names here? seems like we'll have tons of duplicate code if we continue. could we maybe go the route of making the computed settings as properties of the settings class? Then at least it's a uniform import of the settings object everywhere.

Reply:
I think we should be better off with something like a `@dataclass` that holds the default params; then you don't have to instantiate the object and can simply pull the attributes. it looks much clearer & cleaner to keep everything into separate data classes; like, BOARD_GAP_STR can part of a new dataclass that can inherit from Settings. Or we can rename Settings -> DefaultSettings, and then create a `class Settings(DefaultSettings)`.

Reply:
I'm ok with having multiple settings classes, but I don't see why a dataclass solves the issue. The pydantic settings class is already equivalent to a dataclass (as pydantic models are), and has the additional benefit that all members are settable via environment variables out of the box.
I'm also generally not a fan of importing individual symbols, and I'd rather have them namespaced (i.e. I'd rather have `settings.x` instead of `x` so I can see directly where `x` comes from.
If the verbosity of `from arc.settings import settings` is too much, we can do some renaming so that we can import the object with `from arc import settings`.

Reply:
the pydantic skips some of the functionality of a dataclass: with a dataclass you can do `Settings.x`; with, pydantic, you gotta do `Settings().x`, but that's bad because every time you would do this you instantiate a new object in memory... so the other way is to do something like `settings = Settings()` to properly use `settings.x`, which looks a little bit mouthful when you can do simply `Settings.x`. this is basically the reason I mentioned a dataclass for this purpose.